最新源码无法编译qmi_wwan_f，qmi_wwan

```log   
              from /home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:13:
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c: In function 'qmap_register_device':
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1170:21: warning: passing argument 1 of '__builtin_memcpy' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
 1170 |     memcpy (qmap_net->dev_addr, real_dev->dev_addr, ETH_ALEN);
      |             ~~~~~~~~^~~~~~~~~~
./include/linux/fortify-string.h:469:27: note: in definition of macro '__fortify_memcpy_chk'
  469 |         __underlying_##op(p, q, __fortify_size);                        \
      |                           ^
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1170:5: note: in expansion of macro 'memcpy'
 1170 |     memcpy (qmap_net->dev_addr, real_dev->dev_addr, ETH_ALEN);
      |     ^~~~~~
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1170:21: note: expected 'void *' but argument is of type 'const unsigned char *'
 1170 |     memcpy (qmap_net->dev_addr, real_dev->dev_addr, ETH_ALEN);
      |             ~~~~~~~~^~~~~~~~~~
./include/linux/fortify-string.h:469:27: note: in definition of macro '__fortify_memcpy_chk'
  469 |         __underlying_##op(p, q, __fortify_size);                        \
      |                           ^
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1170:5: note: in expansion of macro 'memcpy'
 1170 |     memcpy (qmap_net->dev_addr, real_dev->dev_addr, ETH_ALEN);
      |     ^~~~~~
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c: In function 'qmi_wwan_bind':
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1860:31: error: assignment of read-only location '*dev->net->dev_addr'
 1860 |         dev->net->dev_addr[0] |= 0x02;    /* set local assignment bit */
      |                               ^~
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:1861:31: error: assignment of read-only location '*dev->net->dev_addr'
 1861 |         dev->net->dev_addr[0] &= 0xbf;    /* clear "IP" bit */
      |                               ^~
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c: In function 'rmnet_usb_bind':
/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.c:2072:5: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
 2072 |     int status = qmi_wwan_bind(dev, intf);
      |     ^~~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:250: /home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/qmi_wwan_f.o] Error 1
make[4]: *** [Makefile:2012: /home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0] Error 2
make[4]: Leaving directory '/home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/linux-6.1.35'
make[3]: *** [Makefile:50: /home/yoki/lede/build_dir/target-x86_64_musl/linux-x86_64/qmi_wwan_f-1.0/.built] Error 2
make[3]: Leaving directory '/home/yoki/lede/package/wwan/driver/fibocom_QMI_WWAN'
time: package/wwan/driver/fibocom_QMI_WWAN/compile#0.59#1.07#2.38
    ERROR: package/wwan/driver/fibocom_QMI_WWAN failed to build.
make[2]: *** [package/Makefile:116: package/wwan/driver/fibocom_QMI_WWAN/compile] Error 1
make[2]: Leaving directory '/home/yoki/lede'
make[1]: *** [package/Makefile:110: /home/yoki/lede/staging_dir/target-x86_64_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/yoki/lede'
make: *** [/home/yoki/lede/include/toplevel.mk:231: world] Error 2
```

--- a/package/wwan/driver/fibocom_QMI_WWAN/src/qmi_wwan_f.c
+++ b/package/wwan/driver/fibocom_QMI_WWAN/src/qmi_wwan_f.c
@@ -281,8 +281,11 @@ static int bridge_arp_reply(struct net_device *net, struct sk_buff *skb, uint br
             __skb_pull(reply, skb_network_offset(reply));
             reply->ip_summed = CHECKSUM_UNNECESSARY;
             reply->pkt_type = PACKET_HOST;
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
             netif_rx_ni(reply);
+#else
+            netif_rx(reply);
+#endif
         }
         return 1;
     }
@@ -650,8 +653,13 @@ static void rmnet_vnd_update_rx_stats(struct net_device *net,
     struct pcpu_sw_netstats *stats64 = this_cpu_ptr(dev->stats64);
 
     u64_stats_update_begin(&stats64->syncp);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
     stats64->rx_packets += rx_packets;
     stats64->rx_bytes += rx_bytes;
+#else
+	u64_stats_add(&stats64->rx_packets, rx_packets);
+	u64_stats_add(&stats64->rx_bytes, rx_bytes);
+#endif
     u64_stats_update_end(&stats64->syncp);
 #else
     net->stats.rx_packets += rx_packets;
@@ -666,8 +674,13 @@ static void rmnet_vnd_update_tx_stats(struct net_device *net,
     struct pcpu_sw_netstats *stats64 = this_cpu_ptr(dev->stats64);
 
     u64_stats_update_begin(&stats64->syncp);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
     stats64->tx_packets += tx_packets;
     stats64->tx_bytes += tx_bytes;
+#else
+    u64_stats_add(&stats64->tx_packets, tx_packets);
+    u64_stats_add(&stats64->tx_bytes, tx_bytes);
+#endif
     u64_stats_update_end(&stats64->syncp);
 #else
     net->stats.tx_packets += tx_packets;
@@ -700,10 +713,17 @@ static struct rtnl_link_stats64 *_rmnet_vnd_get_stats64(struct net_device *net, 
 
         do {
             start = u64_stats_fetch_begin_irq(&stats64->syncp);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
             rx_packets = stats64->rx_packets;
             rx_bytes = stats64->rx_bytes;
             tx_packets = stats64->tx_packets;
             tx_bytes = stats64->tx_bytes;
+#else
+	    rx_packets = u64_stats_read(&stats64->rx_packets);
+	    rx_bytes = u64_stats_read(&stats64->rx_bytes);
+	    tx_packets = u64_stats_read(&stats64->tx_packets);
+	    tx_bytes = u64_stats_read(&stats64->tx_bytes);
+#endif
         } while (u64_stats_fetch_retry_irq(&stats64->syncp, start));
 
         stats->rx_packets += rx_packets;
@@ -1857,8 +1877,16 @@ static int qmi_wwan_bind(struct usbnet *dev, struct usb_interface *intf)
 
     /* make MAC addr easily distinguishable from an IP header */
     if (possibly_iphdr(dev->net->dev_addr)) {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,18,0)
         dev->net->dev_addr[0] |= 0x02;    /* set local assignment bit */
         dev->net->dev_addr[0] &= 0xbf;    /* clear "IP" bit */
+#else
+	u8 addr = dev->net->dev_addr[0];
+
+	addr |= 0x02;	/* set local assignment bit */
+	addr &= 0xbf;	/* clear "IP" bit */
+	dev_addr_mod(dev->net, 0, &addr, 1);
+#endif
     }
     if (!_usbnet_get_stats64)
         _usbnet_get_stats64 = dev->net->netdev_ops->ndo_get_stats64;

